ofnetAgent.go

```diff
@@ -565,7 +565,7 @@ func (self *OfnetAgent) AddNetwork(vlanId uint16, vni uint32, Gw string, Vrf str
 func (self *OfnetAgent) RemoveNetwork(vlanId uint16, vni uint32, Gw string, Vrf string) error {
 	// Dont handle endpointDB operations during this time
 	self.lockDB()
-	self.unlockDB()
+	defer self.unlockDB()
```

Comment:
I Vaguely remember there was a deadlocking problem here since EndpointDelete() below tries to take the lock again.. 
you might want to try running system test with this change..